samples/ComponentExamples/src/utils/getEndpointUrl.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+let endpointUrl: string | undefined;
+
+export const getEndpointUrl = async (): Promise<string> => {
+  if (endpointUrl === undefined) {
+    try {
+      const getRequestOptions = {
+        method: 'GET'
+      };
+      const response = await fetch('/getEndpointUrl', getRequestOptions);
+      const retrievedendpointUrl = await response.text().then((endpointUrl) => endpointUrl);
+      endpointUrl = retrievedendpointUrl;
+      return retrievedendpointUrl;
+    } catch (error) {
+      console.error('Failed at getting environment url, Error: ', error);
```

Comment:
nit: Errors should be logged _if and only if_ they are not raised any further. Logging and re-raising leads to duplicate log statements for the same error.